FAQ and Troubleshooting article (#5890)
wix/wix-style-react: docs/usage/Troubleshooting.md

# What should I do when I get in trouble?

In this doc we will share how we troubleshoot issues that we experience when we struggle with a component or a testkit usage:

1. Check whether there is an example that fits your scenario in the [storybook](https://www.wix-style-react.com).
2. Check our [Cheetsheet](https://www.wix-style-react.com/?path=/story/introduction-cheatsheet--components-cheatsheet) which includes a view of most of the library components inventory and links to the relevant documentation pages.
3. Use the [Playground](https://www.wix-style-react.com/?path=/story/introduction-playground--playground) to easily test the component abilities.
4. You may find an answer to your question in our [Frequently Asked Questions](https://github.com/wix/wix-style-react/tree/master/docs/FAQ#frequently-asked-questions) article.

> 5. Each component within the library is tested with our own drivers. This is a great way to document the proper usage of the component and drivers. You can read the tests to learn how the component should be properly used and tested. Checkout the [sourcecode](https://github.com/wix/wix-style-react). Each component's tests files are listed under:

Reviewer: ```suggestion
5. Each component within the library is tested with our own drivers. This is a great way to document the proper usage of the component and drivers. You can read the tests to learn how the component should be properly used and tested. Checkout the [source code](https://github.com/wix/wix-style-react). Each component's tests files are listed under:
```

`src/ComponentName/test/...`.

> 6. Create a new [Yoshi](https://github.com/wix/yoshi) project to serve as a clean environment to test the component - isolate all the noise you got from your own project.

Reviewer: should probably remove this suggestion, as yoshi is not open source anymore

# What should i do when i get in trouble?
# What should I do when I get in trouble?

> When i find my self in times of trouble, this small readme comes to me.

Reviewer 2: ![image](https://user-images.githubusercontent.com/4023882/94120857-4a480600-fe59-11ea-8055-89900d3bca4f.png)

Author: LOL

1. Check whether there is an example that fits your scenario in the [storybook](https://www.wix-style-react.com).
2. Check our [Cheetsheet](https://www.wix-style-react.com/?path=/story/introduction-cheatsheet--components-cheatsheet) which includes a view of most of the library components inventory and links to the relevant documentation pages.
3. Use the [Playground](https://www.wix-style-react.com/?path=/story/introduction-playground--playground) to easily test the component abilities.
4. You may find an answer to your question in our [Frequently Asked Questions](https://github.com/wix/wix-style-react/tree/master/docs/FAQ#frequently-asked-questions) article.

> 5. Each component within the library is tested with our on drivers. This is a great way to document the proper usage of the component and drivers. You can learn from the tests on how the component should properly used / tested. Checkout the [sourcecode](https://github.com/wix/wix-style-react). Each component's tests files are listed under:

Reviewer 2: ```suggestion
5. Each component within the library is tested with our own drivers. This is a great way to document the proper usage of the component and drivers. You can read the tests to learn how the component should be properly used and tested. Checkout the [sourcecode](https://github.com/wix/wix-style-react). Each component's tests files are listed under:
```

`src/ComponentName/test/...`.
6. Create a new [Yoshi](https://github.com/wix/yoshi) project to serve as a clean environment to test the component - isolate all the noise you got from your own project.

> 7. Strip down abilities until you get to the bottom of it. Saying we have for example the `<Table/>` component which does not work as we expect it to. All the above steps did not help us. The thing we would do is one of the two:

Reviewer 2: ```suggestion
7. Strip down abilities until you get to the bottom of it. Let's say we have the `<Table/>` component which does not work as expected. All the above steps did not help us. We could try one of the following:
```
